### PR TITLE
[ENH] Remove the uses of the nested univariate data generator

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### What does this implement/fix? Explain your changes.

A lot of the tests currently create an example classification problem in the nested_univ format using methods buried deep in testing. This replaces that. Its got bigger than I thought it would, there is a fair amount of spaghetti to wade through. 

1. Implements numpy and list generators, taken directly from here
https://github.com/time-series-machine-learning/tsml-py/blob/main/tsml/utils/testing.py#L129
2. Refactors make_classification_problem, now have three explicit problem makers by data type.
3. Removes all use of _make_Panel_X
4. Removes the use of nested univ data generators from distances testing functions. Currently it is running all tests with both nested_univ and numpy3D, which is pointless because its all converted in the base classes. This should give a little testing speed up in test_all_estimators  .... saves 30 seconds, every little helps :)
5. Removes the use of nested univ from testing scenarios for classification and clustering testing
6. Uses list of numpy format for unequal length testing instead of dataframe for testing scenarios

### CODE REVIEW

File: aeon/utils/_testing/collection.py [hunk 1]
```diff
@@ -1,21 +1,215 @@
...
+__author__ = ["mloning", "fkiraly", "TonyBagnall", "MatthewMiddlehurst"]
 __all__ = [
-    "make_classification_problem",
-    "make_regression_problem",
+    "make_3d_test_data",
+    "make_2d_test_data",
+    "make_unequal_length_test_data",
+    "make_nested_df_classification_data",
+    "make_nested_dataframe_regression_problem",
     "make_transformer_problem",
```
Comment (MatthewMiddlehurst) [inline aeon/utils/_testing/collection.py:12]:
```suggestion
    "make_3d_numpy_data",
    "make_2d_numpy_data",
    "make_numpy_list_data",
    "make_nested_df_classification_data",
    "make_nested_df_regression_data",
    "make_transformer_problem",
```

Naming could be a bit more descriptive. Nested DF one could be a single function with a parameter, probably.

**Review by MatthewMiddlehurst — DISMISSED** [review]

File: aeon/classification/tests/test_sklearn_compatability.py [hunk 2]
```diff
@@ -36,12 +36,7 @@
 
 from aeon.classification.interval_based import CanonicalIntervalForest
 from aeon.transformations.panel.interpolate import TSInterpolator
-from aeon.utils._testing.panel import _make_panel_X, make_classification_problem
-
-DATA_ARGS = [
-    {"return_numpy": True, "n_columns": 2},
-    {"return_numpy": False, "n_columns": 2},
-]
```
[the remark this answers is not in the document]
Comment (MatthewMiddlehurst) [reply aeon/classification/tests/test_sklearn_compatability.py:44]: This doesn't have to be done in the same PR if its a lot of work. Have to go through the test again anyway really.

File: aeon/utils/_testing/collection.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/utils/_testing/collection.py:12]: long names cause problems with text wrapping, but I'll have a go.

Comment (TonyBagnall) [reply aeon/utils/_testing/collection.py]: hopefully sorted it out

File: aeon/datatypes/tests/test_series_to_panel_converters.py [hunk 3]
```diff
@@ -38,21 +38,10 @@ def test_convert_numpy_panel_to_series():
 
 def test_convert_df_series_to_panel():
     """Test output format of series-to-panel for dataframe type input."""
-    X_series = _make_series(n_columns=2, return_numpy=False)
+    X_series, _ = make_nested_dataframe_data(n_cases=2)
```
Comment (MatthewMiddlehurst) [inline aeon/datatypes/tests/test_series_to_panel_converters.py:41]: does this not break the test? _make_series would just be a single series right

File: aeon/datatypes/tests/test_series_to_panel_converters.py [hunk 4]
```diff
@@ -38,21 +38,10 @@ def test_convert_numpy_panel_to_series():
...
-
-def test_convert_df_panel_to_series():
-    """Test output format of panel-to-series for dataframe type input."""
-    X_panel = _make_panel(n_instances=1, n_columns=2, return_mtype="pd-multiindex")
-
-    X_series = convert_Panel_to_Series(X_panel)
-
-    assert isinstance(X_series, pd.DataFrame)
-    assert len(X_series) == len(X_panel)
-    assert (X_series.values == X_panel.values).all()
```
Comment (MatthewMiddlehurst) [inline aeon/datatypes/tests/test_series_to_panel_converters.py:58]: Why is this being removed? is it redundant with another test?

Comment (TonyBagnall) [reply aeon/datatypes/tests/test_series_to_panel_converters.py:58]: good question, I'll take a look

Comment (TonyBagnall) [reply aeon/datatypes/tests/test_series_to_panel_converters.py:58]: weird, its still there, I think I must have moved it

File: aeon/datatypes/tests/test_series_to_panel_converters.py (hunk 3 above)
Comment (TonyBagnall) [reply aeon/datatypes/tests/test_series_to_panel_converters.py]: its doesnt break the test, but god knows why! I have changed it back

**Review by MatthewMiddlehurst — DISMISSED** [review]

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**